### ISSUE TITLE
macOS: recording right ⌘ as hotkey saves generic Command — left ⌘ also triggers, and the app's own synthetic paste restarts recording

### ISSUE THREAD

**Issue by symphonix84** [freestyle-voice/freestyle#448]
### Summary

Recording a **right-side modifier** (e.g. right ⌘) as a modifier-only push-to-talk hotkey silently saves the *generic* modifier category (`Command`) instead of the side-specific one (`RightCommand`). This causes two user-visible bugs on macOS:

1. **Both modifier keys trigger dictation.** The user picked right ⌘ only, but left ⌘ starts recording too, including during everyday shortcuts.
2. **Recording ghost-restarts right after hold-release.** On release, the transcript pastes correctly — but the pill closes and reopens a split second later and keeps recording. Freestyle is triggering itself: `macos-fast-paste` posts a synthetic Cmd+V (`macos-fast-paste.swift:60-64`), the key listener's CGEvent tap sees that event and emits `FLAGS:command` (`macos-key-listener.swift:402`), and a `Command` modifier-only hotkey matches it (`key-listener.ts` → `checkMacHotkeyMatch`). Nothing releases it afterwards, so recording continues until the next keystroke.

### Reproduction (v0.5.0, macOS arm64)

1. Settings → change the push-to-talk hotkey, press **right ⌘**. The saved value is `Command` (visible in `settings` table / the `macos-key-listener Command` process arg).
2. Press **left ⌘** → recording starts (bug 1).
3. Hold **right ⌘**, dictate, release → text pastes, then the pill immediately reopens and keeps recording (bug 2).

### Root cause

`hotkey-recorder.ts` deliberately collapses right modifiers to their generic category:

```ts
const RIGHT_MODIFIER_KEYS: Record<string, string> = {
  RightOption: "Alt",
  ...
  RightCommand: "Command",
  ...
};
```

so the settings UI can only ever save `Command`/`Alt`/… for a modifier-only capture — even though the runtime matcher **already fully supports side-specific hotkeys**: `checkMacHotkeyMatch` has a direct `macModState.has(...)` path for `RightCommand`/`RightOption`/etc., `normalizeAccelerator` and `isValidAccelerator` both accept them, and the native listener emits `RIGHT_MOD_DOWN/UP:RightCommand`.

I verified against the real matcher (driving `NativeKeyListener.handleLine` like `tests/key-listener.test.ts` does):

- hotkey `Command`: `FLAGS:command` alone fires `onKeyDown` → left ⌘ triggers, and the synthetic paste re-triggers (`down, up, down`).
- hotkey `RightCommand`: left ⌘ is ignored, right ⌘ hold/release works, and the synthetic Cmd+V does **not** re-trigger.

Manually writing `RightCommand` into the settings DB fixes both symptoms immediately, confirming the diagnosis end-to-end.

### Suggested fixes

1. **Recorder:** preserve side-specificity for modifier-only captures — when the recorded combo is just a right-side modifier, save `RightCommand`/`RightOption`/… instead of the generic category (chords like right-⌘+Space can keep collapsing to `Command+Space`, matching current matcher semantics).
2. **Self-trigger hardening (separate, smaller):** tag the synthetic paste events with a private `CGEventSource` user-data marker in `macos-fast-paste.swift` and skip them in `macos-key-listener.swift`, so even a deliberately-chosen generic `Command` hotkey can't be re-triggered by Freestyle's own paste.

Happy to submit a PR for both — wanted to file the diagnosis first in case you have preferences on the approach (especially whether modifier-only generic captures should remain possible at all).
